openlibrary/catalog/add_book/__init__.py

```diff
@@ -678,10 +677,369 @@ def load_data(rec, account_key=None):
+    :param overwrite: If True, use source_rec values to overwrite those in edition.
+    :return: A tuple containing the (possibly modified) edition and a bool indicating
+        whether the edition needs to be saved.
+    """
+    # Update `edition` str and int value fields with `rec_edition` values. Don't overwrite
+    # existing fields by default.
+    for field, value in source_rec.items():
+        if field in edition and overwrite is False:
+            continue
+
+        if isinstance(value, (str, int)):
+            edition[field] = value
+            need_edition_save = True
+            continue
+
+        if isinstance(value, dict):
+            if value.get('type') in ('/type/int', '/type/text'):
+                edition[field] = value
+                need_edition_save = True
+                continue
+
+    return edition, need_edition_save
+
+
+# Add list fields to edition as needed.
+# Fields in this list must be the same format in both `rec` and `edition`.
+# For example, if the `edition` field value is a `Thing`, the fields will
+# not match properly.
+EDITION_FIELDS: Final = [
+    'dewey_decimal_class',
+    'isbn_10',
+    'isbn_13',
+    'lc_classifications',
+    'lccn',
+    'local_id',
+    'oclc_numbers',
+    'publish_places',
+    'publishers',
+    'series',
+    'source_records',
+    'translation_of',
+    'work_titles',
+]
+
+
+def add_missing_lists_and_list_values_to_edition(
+    source_rec: dict,
+    edition: Edition,
+    edition_fields: list[str],
+    need_edition_save: bool,
+    overwrite: bool = False,
+) -> tuple[Edition, bool]:
+    """
+    Add missing lists and list values from source_rec to the target edition.
+
+    :param source_rec: A `rec` dictionary *returned from build_query()*.
+    :param edition: The matched target edition that will be updated with any new values.
+    :param edition_fields: A list of field names, themselves containing lists, to copy.
+    :param need_edition_save: True if the edition has been modified and needs saving.
+    :param overwrite: If True, use source_rec values to overwrite those in edition.
+    :return: A tuple containing the (possibly modified) edition and a bool indicating
+        whether the edition needs to be saved.
+    """
+    for field in edition_fields:
+        # Skip non-existent or empty rec fields.
+        if not source_rec.get(field):
+            continue
+
+        # Re-write specific fields in *source_rec* to scrub data before (potential)
+        # copying to edition.
+        if field == 'publish_places':
+            # Places such as "[Paris]" should be stripped.
+            source_rec[field] = [place.strip("][)(") for place in source_rec[field]]
+        elif field == 'series':
+            # The Series field may have volumes listed that need removal, e.g.:
+            # "Folio, Policier : roman noir -- 820"
+            source_rec[field] = [
+                series.split("--")[0].strip() for series in source_rec[field]
```

Comment:
I may have steered you wrong here. The problem could really be a poorly named field rather than mismatched data. Series and volume are arguably both useful information and since there's no structured place to put the volume, tacking it onto the series name as has been done historically makes sense. You should check with  @seabelis & @hornc for their preference.

If you do need to split them, the " -- " is added by the MARC import code, so you could ask @hornc to return the two pieces of info separately rather than joining then splitting.

Reply:
I think my ignorance is more to blame here than anything. I should have thought about from where the data come. Had I looked at the MARC data I'd have seen:
```
<datafield tag="490" ind1="0" ind2=" ">
  <subfield code="a">Folio, Policier : roman noir ;</subfield>
  <subfield code="v">820</subfield>
</datafield>
```
Then I'd have realized `--` must be added later, and that should have made me realize I had more work to do.

I will follow up with @seabelis and @hornc. Thanks for clarifying this! :)

Reply:
I went ahead and removed this change to the series data, and by extension volume, data. Thank you for the heads up!